### PR TITLE
fix(stepper): revisit horizontal layout approach to avoid rearranging the DOM

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #1258 

## Summary

This revisit the logic the stepper uses to switch between horizontal and vertical layouts. Instead of moving the content around the DOM, the stepper & item styles now use `display: grid` and `display: contents` (for horizontal layout), respectively, to accomplish the same layout.

### Notes

- `display: contents` takes the host element out of tab order, so some redundant code and styles needs to be applied to the focusable element on items when `layout=horizontal`

### Pending

* [x] update styles to have consistent spacing
* [ ] ~add a11y attributes to `stepper-item` focusable element when `layout=horizontal`~ no longer needed
* [x] update grid definition styles to accommodate any number of items
* [x] update E2E tests
* [x] DRY up CSS

### CODE REVIEW

File: src/components/stepper-item/stepper-item.tsx [hunk 1]
```diff
@@ -135,7 +133,13 @@ export class StepperItem implements InteractiveComponent {
     return (
       <Host aria-expanded={this.active.toString()} onClick={() => this.emitRequestedItem()}>
         <div class="container">
-          <div class="stepper-item-header">
+          <div
+            class="stepper-item-header"
+            tabIndex={
+              /* additional tab index logic needed because of display: contents */
```
Comment (driskull) [inline src/components/stepper-item/stepper-item.tsx:139]: This part is a bit weird...

In the horizontal case, the content isn't within the "button" whereas in the vertical case, the content is within the "button". That seems like kind of a big difference. Can we make them consistent? The content is within the trigger area in one layout and not in the other.

Comment (driskull) [reply src/components/stepper-item/stepper-item.tsx:139]: Might be a breaking change but worth it.

**Review by driskull — COMMENTED** [review]
Yay! this is like a gazillion times better 👍

File: src/components/stepper-item/stepper-item.tsx (hunk 1 above)
Comment (jcfranco) [reply src/components/stepper-item/stepper-item.tsx:167]: Sorry for the delayed reply. I agree that both layouts having different structure is odd. In the meantime, I think we should look into addressing the immediate issue and we can discuss redesigning this. cc @macandcheese @ashetland

**Comment by jcfranco** [comment]
@macandcheese @asangma I know this week is a bit hectic with UC going on, but can either of you help look at the styles and layout changes? The reported diffs from Screener seem to be laid out better, at least to me.

**Comment by macandcheese** [comment]
> @macandcheese @asangma I know this week is a bit hectic with UC going on, but can either of you help look at the styles and layout changes? The reported diffs from Screener seem to be laid out better, at least to me.

Yep - agreed the changes look better, the updated css also equalizes the width of step items which is an improvement. Thanks @jcfranco

File: src/components/stepper-item/stepper-item.tsx [hunk 2]
```diff
@@ -191,7 +200,17 @@ export class StepperItem implements InteractiveComponent {
...
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(): Promise<void> {
+    (this.layout === "vertical" ? this.el : this.headerEl).focus();
```
Comment (driskull) [inline src/components/stepper-item/stepper-item.tsx:213]: Maybe...

```
(this.layout === "vertical" ? this.el : this.headerEl)?.focus();
```

just to be safe.

**Review by driskull — APPROVED** [review]
🪄 Awesome 🪄  🌈 🐰

File: src/components/stepper-item/stepper-item.scss [hunk 3]
```diff
@@ -245,6 +303,28 @@
   @apply border-color-danger-hover;
 }
 
+:host([layout="horizontal"][complete]) .stepper-item-header {
+  border-color: rgba($h-bb-060, 0.5);
+}
+:host([layout="horizontal"][complete]:hover:not([disabled]):not([active])) .stepper-item-header,
```
[the remark this answers is not in the document]
Comment (jcfranco) [reply src/components/stepper-item/stepper-item.scss:309]: I like this idea very much. Maybe stylelint's [selector-max-compound-selectors rule](https://stylelint.io/user-guide/rules/list/selector-max-compound-selectors/) + custom message can help?

[the remark this answers is not in the document]
Comment (driskull) [reply src/components/stepper-item/stepper-item.scss:309]: Yeah that would be nice.

[the remark this answers is not in the document]
Comment (jcfranco) [reply src/components/stepper-item/stepper-item.scss:299]: My original suggestion didn't work, but it looks like we could use [`selector-max-specificity`](https://stylelint.io/user-guide/rules/list/selector-max-specificity) instead. We'd just need to come up with a sensible value for the max specificity. I'll submit a PR for this.

**Pull request merged**